pkg/hive/hiveconfig.go

```diff
@@ -0,0 +1,247 @@
+	runtimeClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// HiveConfigBuilder provides struct for the HiveConfig object containing connection to
+// the cluster and the HiveConfig definitions.
+type HiveConfigBuilder struct {
+	Definition *hiveV1.HiveConfig
+	Object     *hiveV1.HiveConfig
+	errorMsg   string
+	apiClient  runtimeClient.Client
+}
+
+// HiveConfigAdditionalOptions additional options for HiveConfig object.
+type HiveConfigAdditionalOptions func(builder *HiveConfigBuilder) (*HiveConfigBuilder, error)
+
+// NewHiveConfigBuilder creates a new instance of HiveConfigBuilder.
+func NewHiveConfigBuilder(apiClient *clients.Settings, name string) *HiveConfigBuilder {
+	glog.V(100).Infof(
+		`Initializing new HiveConfig structure with the following params: name: %s`, name)
+
+	builder := HiveConfigBuilder{
+		apiClient: apiClient.Client,
+		Definition: &hiveV1.HiveConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+			Spec: hiveV1.HiveConfigSpec{},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the HiveConfig is empty")
+
+		builder.errorMsg = "hiveconfig 'name' cannot be empty"
+	}
+
+	return &builder
+}
+
+// WithOptions creates ClusterDeployment with generic mutation options.
+func (builder *HiveConfigBuilder) WithOptions(
+	options ...HiveConfigAdditionalOptions) *HiveConfigBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Setting HiveConfig additional options")
+
+	for _, option := range options {
+		if option != nil {
+			builder, err := option(builder)
+
+			if err != nil {
+				glog.V(100).Infof("Error occurred in mutation function")
+
+				builder.errorMsg = err.Error()
+
+				return builder
+			}
+		}
+	}
+
+	return builder
+}
+
+// PullHiveConfig loads an existing HiveConfig into HiveConfigBuilder struct.
+func PullHiveConfig(apiClient *clients.Settings, name string) (*HiveConfigBuilder, error) {
+	glog.V(100).Infof("Pulling existing HiveConfig name: %s", name)
+
+	builder := HiveConfigBuilder{
+		apiClient: apiClient.Client,
+		Definition: &hiveV1.HiveConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+		},
+	}
+
+	if name == "" {
+		builder.errorMsg = "hiveconfig 'name' cannot be empty"
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("hiveconfig object %s doesn't exist", name)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Get fetches the defined HiveConfig from the cluster.
+func (builder *HiveConfigBuilder) Get() (*hiveV1.HiveConfig, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Getting HiveConfig %s", builder.Definition.Name)
+
+	HiveConfig := &hiveV1.HiveConfig{}
+	err := builder.apiClient.Get(context.TODO(), runtimeClient.ObjectKey{
+		Name: builder.Definition.Name,
+	}, HiveConfig)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return HiveConfig, err
+}
+
+// Create generates a HiveConfig on the cluster.
```

Comment:
Note that hiveconfig gets autocreated.
If you delete it - it gets autocreated.
Do we need this method?

Reply:
I was also wondering about that but couldn't find anything that stood out in their repo. I think we can remove it as I have never had to manually create one

Reply:
@trewest testing

Reply:
@ElaiShalevRH You should be safe to remove the Create method. Just change your update method to align with this one: https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/deployment/deployment.go#L395